modules/internal/ChapelDomain.chpl

```diff
@@ -1283,6 +1283,271 @@ module ChapelDomain {
+
+        // Check to make sure value and array element types match.
+        if arr.eltType != value.type then
+          compilerError('Initialization value type \'' + value:string +
+                        '\' does not match array element type \'' +
+                        arr.eltType:string + '\'');
+
+        // Only certain forms of index are permissable.
+        _checkThatIndexMatchesArrayShape(arr, idx);
+
+        // TODO: Is this even worth checking? It's O(n) number of arrays.
+        if !_isArrayOwnedByLhsDomain(arr) then
+          halt('Can only initialize elements of arrays declared over ' +
+               'the domain being resized');
+
+        if !arr.domain.contains(idx) then
+          halt('Domain index out of bounds: ' + idx:string);
+
+        // Get a reference to the array slot.
+        ref elem = arr[idx];
+
+        if !isElementInitialized(arr, idx) {
+          _moveInitializeElement(arr, idx, value);
+        } else {
+          elem = value;
+        }
+      }
+
+      pragma "no doc"
+      proc enterThis() ref {
+
+        // TODO: Is it possible to nest unsafe assignments? Future work...
+        if _isActiveManager {
+          halt('Cannot nest a manager for unsafe domain assignment');
+        } else {
+          _isActiveManager = true;
+        }
+
+        for baseArr in _lhsBaseArrays() {
+
+          // Don't care about arrays with default-initializable elements.
+          if baseArr.chpl_isElementTypeDefaultInitializable() then
+            continue;
+
+          // Array elements are non-nilable classes. In this case, tell
+          // the array to clear new slots when resizing by setting the
+          // resize policy. We can use 'nil' slots as sentinel values.
+          if baseArr.chpl_isElementTypeNonNilableClass() {
+            const policy = if _checks then chpl_DdataResizePolicy.ClearMem
+                                      else chpl_DdataResizePolicy.SkipInit;
+            baseArr.chpl_setResizePolicy(policy);
+
+          // The array element type is non-default-initializable (but it is
+          // not a non-nilable class).
+          } else {
+
+            // TODO: Future work could handle this case by using a bitvector
+            // to mark slot initialization status. Initialization calls
+            // would have to go through the 'initialize' method in order
+            // for this approach to work.
+            if _checks {
+              halt('Runtime checks are currently only supported for ' +
+                   'arrays of non-nilable classes');
+            } else {
+              const policy = chpl_DdataResizePolicy.SkipInit;
+              baseArr.chpl_setResizePolicy(policy);
+            }
+          }
+        }
+
+        // Create temporary shells for the LHS and RHS domains.
+        var lhsTmpDom = new _domain(_lhsPid, _lhsInstance, _unowned=true);
+        var rhsTmpDom = new _domain(_rhsPid, _rhsInstance, _unowned=true);
```

Comment:
Just so I'm understanding, this is to be able to simply do `lhsTmpDom = rhsTmpDom`, right?

Reply:
Yeah.

Reply:
Thanks!